UnityProject/Assets/Scripts/Communications/SignalEmitter.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections;
+using System.Collections.Generic;
+using Managers;
+using ScriptableObjects.Communications;
+using UnityEngine;
+using Mirror;
+
+namespace Communications
+{
+	public class SignalEmitter : NetworkBehaviour
+	{
+		[SerializeField] private SignalDataSO signalData;
+		private float frequency = 122f;
+
+		public float Frequency
+		{
+			get => frequency;
+			set => frequency = value;
+		}
+
+		/// <summary>
+		/// Tells the SignalManager to send a signal to a receiver
+		/// </summary>
+		public virtual void SendSignal()
```

Comment:
I normally dislike methods were you have to remember to do call ``Base`` before your own logic. Could we instead make the two methods children of this class are supposed to implement ``abstract`` and interface with the manager using another method that will call the abstract method.